### PR TITLE
feat: add simple account abstraction example

### PR DESCRIPTION
**Pull request by avasisht23**
Screen recording and context: https://alchemyinsights.slack.com/archives/C05KFT0E4U9/p1696917521459789

### CODE REVIEW

**Comment by avasisht23** [comment]
Current dependencies on/for this PR:
* main
  * **PR #86** <a href="https://app.graphite.dev/github/pr/OMGWINNING/aa-sdk-staging/86" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/OMGWINNING/aa-sdk-staging/86?utm_source=stack-comment).

File: examples/aa-simple-dapp/src/hooks/useMagicSigner.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { chain, magicApiKey } from "@/config/client";
```
Comment (rthomare) [inline examples/aa-simple-dapp/src/hooks/useMagicSigner.ts:1]: this is cool

File: examples/aa-simple-dapp/src/hooks/useAlchemyProvider.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import {
+  NEXT_PUBLIC_ALCHEMY_GAS_MANAGER_POLICY_ID,
+  NEXT_PUBLIC_ALCHEMY_RPC_URL,
+  chain,
+  simpleAccountFactoryAddress,
+} from "@/config/client";
+// import { getRpcUrl } from "@/config/rpc"; // todo(ajay): use server side
```
Comment (rthomare) [inline examples/aa-simple-dapp/src/hooks/useAlchemyProvider.ts:7]: rm?

File: examples/aa-simple-dapp/src/hooks/useAlchemyProvider.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            rpcClient,
+            owner: signer,
+            chain,
+            entryPointAddress,
+            factoryAddress: simpleAccountFactoryAddress,
+            accountAddress: account,
+          });
+        })
+        .withAlchemyGasManager({
+          policyId: NEXT_PUBLIC_ALCHEMY_GAS_MANAGER_POLICY_ID, // todo(ajay): use server side
```
Comment (rthomare) [inline examples/aa-simple-dapp/src/hooks/useAlchemyProvider.ts:45]: what do you mean server side? FWIW I think just doing it as a local env var is fine.

File: examples/aa-simple-dapp/src/hooks/useAlchemyProvider.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+};
+
+export const useAlchemyProvider = ({
+  entryPointAddress,
+}: AlchemyProviderProps) => {
+  const [provider, setProvider] = useState<AlchemyProvider>(
+    new AlchemyProvider({
+      chain,
+      entryPointAddress,
+      rpcUrl: NEXT_PUBLIC_ALCHEMY_RPC_URL!, // todo(ajay): getRpcUrl(chain.id),
```
Comment (rthomare) [inline examples/aa-simple-dapp/src/hooks/useAlchemyProvider.ts:27]: Let's not ship an example that give people the wrong way to do this. We should also make sure for our own internal classes we don't show people to do it this way either.

Comment (rthomare) [reply examples/aa-simple-dapp/src/hooks/useAlchemyProvider.ts:27]: Going to block on making sure we proxy through our backend

File: examples/aa-simple-dapp/src/config/rpc.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+export function getRpcUrl(chainId: number) {
```
Comment (rthomare) [inline examples/aa-simple-dapp/src/config/rpc.ts:1]: This isn't used right? Can we remove this? Also we should just be sending through our own proxy. Also for this example I'm sure it's find to just use one chain.

File: examples/aa-simple-dapp/src/config/client.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { clientEnv } from "@/env/client.mjs";
+import { sepolia } from "wagmi/chains";
+
+export const entryPointAddress = "0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789";
+export const simpleAccountFactoryAddress =
```
Comment (rthomare) [inline examples/aa-simple-dapp/src/config/client.ts:5]: Why not use the default light account here? Can we use that instead? I feel like we should be pushing people to use light account for everything and not simple account.

File: examples/aa-simple-dapp/src/components/Hero.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
+"use client";
```
Comment (rthomare) [inline examples/aa-simple-dapp/src/components/Hero.tsx:1]: Left some comments in slack about this but TL;DR
- Can we move the AccountKit Docs Upsell to below the title
- Can we give the user some indication of what is happening i.e. we are deploying the smart contract account, now claiming the token.
- Can we also show the smart account address and link it to etherscan?

File: examples/aa-simple-dapp/src/app/globals.css [hunk 8]
```diff
@@ -0,0 +1,3 @@
+@tailwind base;
```
Comment (rthomare) [inline examples/aa-simple-dapp/src/app/globals.css:1]: Why tailwind over chakra?

**Review by rthomare — CHANGES_REQUESTED** [review]

File: examples/aa-simple-dapp/src/components/Hero.tsx (hunk 7 above)
Comment (rthomare) [reply examples/aa-simple-dapp/src/components/Hero.tsx:1]: Also let's not call it Alchemy Token please 😅

File: examples/aa-simple-dapp/src/config/rpc.ts (hunk 5 above)
Comment (avasisht23) [reply examples/aa-simple-dapp/src/config/rpc.ts:1]: doesn't this return the proxy url `api/rpc`? I ripped this from our webapp which seemed to work fine there.

File: examples/aa-simple-dapp/src/config/client.ts (hunk 6 above)
Comment (avasisht23) [reply examples/aa-simple-dapp/src/config/client.ts:5]: using LA now

File: examples/aa-simple-dapp/src/hooks/useAlchemyProvider.ts (hunk 4 above)
Comment (avasisht23) [reply examples/aa-simple-dapp/src/hooks/useAlchemyProvider.ts]: resolved

File: examples/aa-simple-dapp/src/config/rpc.ts (hunk 5 above)
Comment (avasisht23) [reply examples/aa-simple-dapp/src/config/rpc.ts]: resolved

File: examples/aa-simple-dapp/src/components/Hero.tsx (hunk 7 above)
Comment (avasisht23) [reply examples/aa-simple-dapp/src/components/Hero.tsx:1]: all resolved

File: examples/aa-simple-dapp/src/config/rpc.ts [hunk 9]
```diff
@@ -0,0 +1,8 @@
+export function getRpcUrl() {
```
Comment (rthomare) [inline examples/aa-simple-dapp/src/config/rpc.ts:1]: nit I'd say we can literally just kill this file and just add a const for "/api/rpc/"

**Review by rthomare — APPROVED** [review]
Am I crazy are we using LightAccount? Otherwise LGTM

File: examples/aa-simple-dapp/src/config/client.ts (hunk 6 above)
Comment (rthomare) [reply examples/aa-simple-dapp/src/config/client.ts:5]: Let's use the useDefaultLightAccountFactoryAddress() function at somepoint here.

File: examples/aa-simple-dapp/src/hooks/useAlchemyProvider.ts [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+      entryPointAddress,
+      rpcUrl: getRpcUrl(),
+    })
+  );
+
+  const connectProviderToAccount = useCallback(
+    (signer: SmartAccountSigner, account?: Address) => {
+      const connectedProvider = provider
+        .connect((provider) => {
+          return new LightSmartContractAccount({
```
Comment (avasisht23) [inline examples/aa-simple-dapp/src/hooks/useAlchemyProvider.ts:32]: @rthomare we use LA here!

**Pull request merged**